### PR TITLE
add pythia8 automatic configuration on bkg part and other bkg dedicated settings

### PR DESCRIPTION
**Pull request by gconesab**

### CODE REVIEW

File: MC/bin/o2dpg_sim_workflow.py [hunk 1]
```diff
@@ -130,14 +132,77 @@ def getDPL_global_options(bigshm=False,nosmallrate=False):
...
+        if ECMSBKG < 0 and EBEAMABKG < 0 and EBEAMBBKG < 0:
+           print('o2dpg_sim_workflow: Error! bkg ECM or Beam Energy not set!!!')
+           exit(1)
 
         CONFKEYBKG=''
         if args.confKeyBkg!= '':
            CONFKEYBKG=' --configKeyValues ' + args.CONFKEYBKG
 
+        # Background PYTHIA configuration
+        BKG_CONFIG_task=createTask(name='genbkgconf')
```
Comment (sawenzel) [inline MC/bin/o2dpg_sim_workflow.py:183]: You probably need to add this task to the list of requirements for `BKGtask` further below.

File: MC/run/PWGGAJE/run_dirgamma_embedding.sh [hunk 2]
```diff
@@ -55,13 +55,15 @@ echo 'Detector acceptance option ' $PARTICLE_ACCEPTANCE
...
-${O2DPG_ROOT}/MC/bin/o2dpg_sim_workflow.py -eCM ${CONFIG_ENERGY} -col pp -gen pythia8 -proc "dirgamma" \
-                                            -ptHatMin ${PTHATMIN} -ptHatMax ${PTHATMAX}                \
-                                            -tf ${NTIMEFRAMES} -ns ${NSIGEVENTS} -e ${SIMENGINE}       \
-                                            -nb ${NBKGEVENTS} --embedding                              \
-                                            -j ${NWORKERS} -mod "--skipModules ZDC"                    \
-                                            -weightPow ${WEIGHTPOW}                                    \
-                                            -trigger "external" -ini "\$O2DPG_ROOT/MC/config/PWGGAJE/ini/trigger_prompt_gamma.ini"
+${O2DPG_ROOT}/MC/bin/o2dpg_sim_workflow.py -eCM ${CONFIG_ENERGY}                                     \
+                                           -nb ${NBKGEVENTS} --embedding                             \
+                                           -colBkg PbPb -genBkg pythia8 -procBkg "none"              \
```
Comment (sawenzel) [inline MC/run/PWGGAJE/run_dirgamma_embedding.sh:60]: what does "none" mean as process? Is it different from `inel`?

File: MC/run/PWGGAJE/run_jets_embedding.sh [hunk 3]
```diff
@@ -15,7 +15,7 @@
 RNDSEED=${RNDSEED:-0}   # [default = 0] time-based random seed
 NSIGEVENTS=${NSIGEVENTS:-2}
 NBKGEVENTS=${NBKGEVENTS:-1}
-NTIMEFRAMES=${NTIMEFRAMES:-5}
+NTIMEFRAMES=${NTIMEFRAMES:-1}
```
Comment (sawenzel) [inline MC/run/PWGGAJE/run_jets_embedding.sh:18]: do you want to commit this change?

File: MC/run/PWGGAJE/run_dirgamma_embedding.sh (hunk 2 above)
Comment (gconesab) [reply MC/run/PWGGAJE/run_dirgamma_embedding.sh:60]: It means no process specified, apparently Pb-Pb Pythia does not need the process

File: MC/bin/o2dpg_sim_workflow.py (hunk 1 above)
Comment (gconesab) [reply MC/bin/o2dpg_sim_workflow.py:183]: do you mean this? workflow['stages'].append(BKG_CONFIG_task) it is below

File: MC/run/PWGGAJE/run_jets_embedding.sh (hunk 3 above)
Comment (gconesab) [reply MC/run/PWGGAJE/run_jets_embedding.sh:18]: It is an example I was testing, I will change it back

File: MC/run/PWGGAJE/run_dirgamma_embedding.sh (hunk 2 above)
Comment (preghenella) [reply MC/run/PWGGAJE/run_dirgamma_embedding.sh:60]: @gconesab 
PbPb needs more than just not setting the process.
I suggest that you use process `heavyion` as a label that adds the required settings for pythia8 heavy-ion machinery

Comment (gconesab) [reply MC/run/PWGGAJE/run_dirgamma_embedding.sh:60]: in mkpy8cfg.py I activate those when Pb-Pb is set, but I agree it is better to specify a HI case, I will do that

File: MC/bin/o2dpg_sim_workflow.py (hunk 1 above)
Comment (sawenzel) [reply MC/bin/o2dpg_sim_workflow.py:183]: No. I believe that BKG_CONFIG_task needs to finish before BKGtask can start, right? So you need to add the `needs=[]` field for BKGtask.

Comment (gconesab) [reply MC/bin/o2dpg_sim_workflow.py:183]: I tried         BKGtask=createTask(name='bkgsim', lab=["GEANT"], needs=BKG_CONFIG_task['name'], cpu=NWORKERS )
but it does not work

**Comment by gconesab** [comment]
Ciao @sawenzel and @preghenella, I think I addressed all your comments, thanks!
I added a new process "heavy_ion" and renamed the one called "heavy" to "heavy_q". Let me know if you agree

**Comment by gconesab** [comment]
Hi guys, any extra comment? If not, I will merge later today.

File: MC/bin/o2dpg_sim_workflow.py (hunk 1 above)
Comment (sawenzel) [reply MC/bin/o2dpg_sim_workflow.py:183]: You need a `list`: `needs=[BKG_CONFIG_task['name']]`

Comment (gconesab) [reply MC/bin/o2dpg_sim_workflow.py:183]: I think this is what I did on the last commit 
        BKGtask=createTask(name='bkgsim', lab=["GEANT"], needs=[BKG_CONFIG_task['name']], cpu=NWORKERS )

Comment (sawenzel) [reply MC/bin/o2dpg_sim_workflow.py:183]: Ok. This was not apparent from these comments. Did you test the whole workflow? Does it run as expected? If yes, I think we can go ahead.

Comment (gconesab) [reply MC/bin/o2dpg_sim_workflow.py:183]: yes, it seems to work, although I did not let it end, too hot my computer

**Pull request merged**